src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -198,18 +206,38 @@ rpmostreed_deployment_generate_variant (OstreeDeployment *deployment,
   g_variant_dict_insert (&dict, "checksum", "s", csum);
   if (rpmostree_origin_is_locally_assembled (origin))
     {
-      const char *parent = ostree_commit_get_parent (commit);
-      g_assert (parent);
-      g_variant_dict_insert (&dict, "base-checksum", "s", parent);
-      sigs = rpmostreed_deployment_gpg_results (repo, rpmostree_origin_get_refspec (origin),
-                                                parent, &gpg_enabled);
+      base_checksum = ostree_commit_get_parent (commit);
+      g_assert (base_checksum);
+      g_variant_dict_insert (&dict, "base-checksum", "s", base_checksum);
+      sigs = rpmostreed_deployment_gpg_results (repo, refspec, base_checksum, &gpg_enabled);
     }
   else
-    sigs = rpmostreed_deployment_gpg_results (repo, rpmostree_origin_get_refspec (origin),
-                                              csum, &gpg_enabled);
+    {
+      base_checksum = csum;
+      sigs = rpmostreed_deployment_gpg_results (repo, refspec, csum, &gpg_enabled);
+    }
+  variant_add_commit_details (&dict, NULL, commit);
+
+  if (!ostree_repo_resolve_rev (repo, refspec, TRUE,
+                                &pending_base_commitrev, error))
```

Comment:
What I would say here is that at the moment, the data I'm trying to add is just looking for any changes in the origin refspec vs deployed checksum - for each deployment independently.

So it "works" with rebases by ignoring it, because it's a new deployment.   I wouldn't say it's broken, but it's showing you information you don't need.  Admittedly it is a bit redundant too if you do an `atomic host upgrade; atomic host status`.